### ISSUE TITLE
Provide direct support for Best-Effort pins?

### ISSUE THREAD

**Issue by kevina** [ipfs/kubo#3121]
Should we provide direct support for creating best-effort pins?  In some ways it might not be necessary because users can just add stuff to the MFS root that they want to keep around but don't want a recursive pin for.

Related to #3069.

**Comment by whyrusleeping** [comment]
Yeah, i'm in support of having best effort pins. Although i think that in order to do that, we might want to first rethink how pins are stored. Take a look at #1580. That bit might be a really nice addition to have before best effort pins because then users would be able to `ipfs pin add --type=best /ipfs/<hash>/path/to/dataset` and be able to still resolve that full path after a gc run.

**Comment by lidel** [comment]
One can do best-effort pinning like this:
1. `ipfs files cp /ipfs/{cid} /mfs-destination` will add DAG to MFS, acting as "implicit" pin that protects from GC
2. `ipfs pin -r {cid}` (or `ipfs dag stat {cid}` – preload blocks without explicitly pinning, use MFS for GC protection)

My understanding is that if you interrupt (2), the (1) will protect blocks that are already fetched.

I am not sure how valuable is adding best-effort mode to `ipfs pin` unless it is the default behavior.
